src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/Events.ts

```diff
@@ -0,0 +1,35 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as psList from 'ps-list';
+import { DebugSession } from 'vscode';
+
+import { RazorLogger } from '../RazorLogger';
+
+export async function onDidTerminateDebugSession(
+  event: DebugSession,
+  logger: RazorLogger,
+  targetPid: number | undefined,
+) {
+  logger.logVerbose('Terminating debugging session...');
+  try {
+    const processes = await psList();
+    if (targetPid) {
+      try {
```

Comment:
All the nesting makes it decently hard to read. I'd suggest breaking some of these levels out and rely on `let` based variables